### PR TITLE
Add value expressions documentation

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@dobe please review

File: blackbox/docs/sql/reference/value_expressions.txt [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+=================
+Value Expressions
+=================
+
+Value expressions are expressions which return a single value.
+They can be used in many contexts of many statements.
+
+A value expression can be:
+
+- A Literal value
```
Comment (seut) [inline blackbox/docs/sql/reference/value_expressions.txt]: use small letters for Literal, also below

File: blackbox/docs/sql/reference/value_expressions.txt [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+- A Column reference
+
+- A parameter reference
+
+- A operator invocation
```
Comment (seut) [inline blackbox/docs/sql/reference/value_expressions.txt]: An operator...

File: blackbox/docs/sql/reference/value_expressions.txt [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+- An array constructor
+
+
+Literal value
+=============
+
+A literal is a notation to represent a value within a statement.
+
+Different types have a different notation. The simplest forms are:
```
Comment (seut) [inline blackbox/docs/sql/reference/value_expressions.txt]: afaik: "...have different notations."

**Pull request merged**